flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/StreamCommonSubGraphBasedOptimizer.scala

```diff
@@ -50,14 +51,23 @@ class StreamCommonSubGraphBasedOptimizer(planner: StreamPlanner)
     val tableConfig = planner.getTableConfig
     // build RelNodeBlock plan
     val sinkBlocks = RelNodeBlockPlanBuilder.buildRelNodeBlockPlan(roots, tableConfig)
+    val miniBatchRequirementChecker = {
+      (node: RelNode) =>
+        node.isInstanceOf[Filter] ||
+        node.isInstanceOf[Project] ||
+        node.isInstanceOf[TableScan] ||
+        (node.isInstanceOf[LogicalUnion] && node.asInstanceOf[LogicalUnion].all)
+    }
     // infer trait properties for sink block
     sinkBlocks.foreach {
       sinkBlock =>
         // don't require update before by default
         sinkBlock.setUpdateBeforeRequired(false)
-
         val miniBatchInterval: MiniBatchInterval =
-          if (tableConfig.get(ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_ENABLED)) {
+          if (
+            tableConfig.get(ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_ENABLED) &&
+            !sinkBlock.containsAll(miniBatchRequirementChecker)
```

Comment:
The method name `containsAll` is hard to understand what it wanted to do without reading the source code. How about something like `fullySatisfy`? 

With the above mentioned suggestion, it will sound like !sinkBlock.fullySatisfy(noMiniBatchRequired).

Reply:
Agreed. Also, I moved the mini-batch skipping logic from `RelnodeBlock` to `StreamCommonSubGraphBasedOptimizer:: shouldSkipMiniBatch`.